ejb3/src/main/java/org/jboss/as/ejb3/tx/OwnableReentrantLock.java

```diff
@@ -52,8 +52,11 @@ public OwnableReentrantLock() {
     }
 
     public void lock(Object owner) {
+        if (owner == null) {
+            throw new IllegalArgumentException("owner cannot be null");
+        }
         synchronized (this.lock) {
-            if (owner == this.owner) {
+            if (owner.equals(this.owner)) {
```

Comment:
Since we don't know the implementation of the owner object, I suggest using java.util.Objects.equals(owner, this.owner) to avoid a potential NPE.